demo/src/screens/ThankScreen/ThankScreen.js

```diff
@@ -1,13 +1,88 @@
-import React from 'react';
+import React, {useMemo} from 'react';
+import {Container, Row, Col, Jumbotron} from 'react-bootstrap';
+import {ProductListing} from '../../components/ProductListing/ProductListing.js';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
 
 /**
+ * @param {!Object} items The items stored in the site state
  * @return {!JSX} Page component where we thank the user
  *     for their purchase(s).
  */
-export function ThankScreen() {
+export const ThankScreenBase = function({items}) {
+  const /** Array<!JSX> */ productListings = useMemo(() => {
+    const ITEMS_TO_DISPLAY = 6;
+    const products = [];
+    const usedItemSet = new Set();
+
+    let i;
+    for (i = 0; i < ITEMS_TO_DISPLAY; ++i) {
+      let maxPrice = 0;
+      let maxPriceItem;
+      let maxPriceID;
+
+      // Finds and pushes the most expensive item that has not yet been used.
+      for (const [itemID, item] of Object.entries(items)) {
+        if (item.cost > maxPrice && !usedItemSet.has(itemID)) {
+          maxPriceItem = item;
+          maxPriceID = itemID;
+          maxPrice = item.cost;
```

Comment:
seems redundant, it should be sufficient to just maxPriceItem.cost (similar to how it's done for maxPriceItem.name). If it can be undefined you could always `maxPriceItem.cost || 0`.